### PR TITLE
adding golang base

### PR DESCRIPTION
**Pull request by sahilsuneja1**
Signed-off-by: Sahil Suneja <sahilsuneja@gmail.com>

### CODE REVIEW

**Comment by lumjjb** [comment]
Awesome! Thanks @sahilsuneja1 

It seems that the development pattern of go is different from the others. Do you think it is possible to make:

a `golang-build-nabla` as well as a `golang-base-nabla`. Where the former is for the first stage of the multistage build and the later is the `FROM scratch` portion?

This issue has been opened separately and will be resolved in the future:

https://github.com/nabla-containers/nabla-base-build/issues/11

File: tests/gotest/helloworld.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package main
+
+import "fmt"
+import "C"
```
Comment (ricarkol) [inline tests/gotest/helloworld.go]: Is this needed?

File: go-base/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ENV PATH="/root/unikernels/nabla-base-build/rumprun/rumprun-solo5/bin:${PATH}"
+ENV CC="x86_64-rumprun-netbsd-gcc" 
+ENV GOROOT="/gorump"
+ENV GOPATH="/gopath"
+ENV GOTOOLDIR="/gorump/pkg/tool/linux_amd64"
+ENV CGO_ENABLED="1" 
+ENV GOOS="rumprun" 
+ENV PATH="/gorump/bin:${PATH}"
+
+
```
Comment (ricarkol) [inline go-base/Dockerfile]: nit: extra empty space

**Review by ricarkol — APPROVED** [review]
lgtm

**Pull request merged**